### PR TITLE
Fix the potential application crash issues

### PR DESCRIPTION
**Pull request by gaoethan**
`get_topic_name` and `rcl_service_get_service_name` may return NULL

Signed-off-by: Ethan Gao <ethan.gao@linux.intel.com>

### CODE REVIEW

File: rclcpp/src/rclcpp/publisher.cpp [hunk 1]
```diff
@@ -170,7 +170,13 @@ PublisherBase::setup_intra_process(
   StoreMessageCallbackT callback,
   const rcl_publisher_options_t & intra_process_options)
 {
-  auto intra_process_topic_name = std::string(this->get_topic_name()) + "/_intra";
+  const char * topic_name = this->get_topic_name();
+  if (topic_name) {
+    auto intra_process_topic_name = std::string(topic_name) + "/_intra";
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/publisher.cpp]: Have you tried to compile the patch locally (and ran the tests)? It look to me that the variable has a local scope and therefore isn't available below where it is being used.

Comment (gaoethan) [reply rclcpp/src/rclcpp/publisher.cpp]: badly, it becomes a local scope within if{}, and I changed it. Now all have passed the build and tests, thanks !

File: rclcpp/src/rclcpp/publisher.cpp [hunk 2]
```diff
@@ -170,7 +170,14 @@ PublisherBase::setup_intra_process(
...
   const rcl_publisher_options_t & intra_process_options)
 {
-  auto intra_process_topic_name = std::string(this->get_topic_name()) + "/_intra";
+  const char * topic_name = this->get_topic_name();
+  std::string intra_process_topic_name = "";
+  if (topic_name) {
+    intra_process_topic_name = std::string(topic_name) + "/_intra";
+  } else {
+    throw std::runtime_error("Get a invalid null topic name");
+  }
```
Comment (dirk-thomas) [inline rclcpp/src/rclcpp/publisher.cpp]: Please use a similar structure here as above. Check for `!topic_name` and conditionally `throw`, Then the `else` case becomes obsolete.

Comment (gaoethan) [reply rclcpp/src/rclcpp/publisher.cpp]: make sense, changed :smile:

**Comment by dirk-thomas** [comment]
I modified the error message slight. Looks good to me now:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3798/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/934/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3135/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3884/)

**Comment by clalancette** [comment]
Builds are all green, @dirk-thomas thinks it is OK, looks OK to me.  Merging.

**Pull request merged**